### PR TITLE
Dmp 3053.media linked table

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
incorporate  media_linked_table into code.
https://tools.hmcts.net/jira/browse/DMP-3053

### CODE REVIEW

**Review by mestebanez — COMMENTED** [review]
Could we add a integration test to prove that the case records get migrated correctly by using the PLSQL function that has been written.

File: src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+DO
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql:1]: As previously mentioned, would be nice to be made a java flyway class, both for readability and testability but I understand why you havent performed this, so please ignore this comment.

File: src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+$do$
+DECLARE
+   case_id   int;
+   courthouse_name   text;
+	temprow record;
+BEGIN
+FOR temprow IN
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql:8]: Nice to make temprow a bit more meaningful e.g. mediaToBeMigrated?

File: src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
+DO
+$do$
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql:2]: We can remove this and add $$DECLARE and END$$;

File: src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+DECLARE
+   case_id   int;
+   courthouse_name   text;
+	temprow record;
+BEGIN
+FOR temprow IN
+        select med_id, ctr_id, UNNEST(case_number) case_number from darts.media where case_number is not null order by 1,2
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql:9]: Although not related to this pr. Should the table definition have a mutual exclusive check constraint:-

CHECK ((case_number IS NULL AND cas_id IS NOT NULL) 
   OR (cas_id IS NOT NULL AND case_number IS NULL))

File: src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    LOOP
+		select cc.cas_id into case_id from darts.court_case cc, darts.courtroom cr where cr.cth_id = cc.cth_id and cc.case_number = temprow.case_number and cr.ctr_id = temprow.ctr_id;
+		if not found then
+			select ch.courthouse_name into courthouse_name from darts.courtroom cr, darts.courthouse ch where cr.cth_id = ch.cth_id and cr.ctr_id= temprow.ctr_id;
+        	INSERT INTO darts.media_linked_case (mlc_id,med_id, courthouse_name, case_number) VALUES (nextval('mlc_seq'),temprow.med_id,courthouse_name, temprow.case_number);
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql:14]: I notice on new add audios through the api we always create and associated a case src/main/java/uk/gov/hmcts/darts/audio/component/impl/AddAudioRequestMapperImpl.java. Is this a requirement for the migrated data also?

Comment (cmagowan-hmcts) [reply src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql:14]: this script wont be run for migrated data, this is purely for whatever info the testers had in the test database to try to copy it over, but if it isn't copied, its not a big deal, its more of an audit table. no cases should be created by this script

File: src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql (hunk 4 above)
Comment (davet1985) [reply src/main/resources/db/migration/postgres/V1_320_1__copyCaseNumbersToMediaLinkedCase.sql:9]: One for @barry-jones50410

**Review by davet1985 — APPROVED** [review]

**Pull request merged**